elixir/apps/web/lib/web/live/settings/dns.ex

```diff
@@ -1,6 +1,100 @@
 defmodule Web.Settings.DNS do
   use Web, :live_view
 
+  alias Domain.Config
+
+  defp pretty_print_addrs([]), do: ""
+  defp pretty_print_addrs(addrs), do: Enum.join(addrs, ", ")
+
+  defp addrs_to_list(nil), do: []
+
+  defp addrs_to_list(addrs_str) do
+    addrs_str
+    |> String.split(",", trim: true)
+    |> Enum.map(&String.trim/1)
+    |> Enum.uniq()
+  end
+
+  defp config_to_params(config) do
+    addrs_str = Enum.join(config.clients_upstream_dns, ", ")
```

Comment:
I think instead of mapping it back and forth it might be better to make an input work like this:
- If the form is empty - just one DNS input field is rendered
- Once all fields are not empty - one more input is added

This will allow users to add as many servers as they want without clicking Add/Remove button, and Ecto should be able to handle it gracefully if field name is something like `dns[]`.